x-pack/plugins/enterprise_search/server/routes/app_search/credentials.ts

```diff
@@ -8,10 +8,32 @@ import { schema } from '@kbn/config-schema';
 
 import { IRouteDependencies } from '../../plugin';
 
+const tokenSchema = schema.oneOf([
+  schema.object({
+    name: schema.string(),
+    type: schema.literal('admin'),
+  }),
+  schema.object({
+    name: schema.string(),
+    type: schema.literal('private'),
+    read: schema.boolean(),
+    write: schema.boolean(),
+    access_all_engines: schema.boolean(),
+    engines: schema.maybe(schema.arrayOf(schema.string())),
+  }),
+  schema.object({
+    name: schema.string(),
+    type: schema.literal('search'),
+    access_all_engines: schema.boolean(),
+    engines: schema.maybe(schema.arrayOf(schema.string())),
+  }),
+]);
+
 export function registerCredentialsRoutes({
   router,
   enterpriseSearchRequestHandler,
 }: IRouteDependencies) {
+  // Credentials API
```

Comment:
I would have assumed everything in this file is the Credentials API.

EDIT: Oh, I see, you're distinguishing it from credentials (plural) from credential (singular) below

Reply:
And also distinguishing from the moderately confusing `/details` route (which I think exists only just to get `engines` data, and I'm hoping we can replace at some point with a call to an engines API)